src/bindings/python/flux/job.py

```diff
@@ -1047,3 +1052,38 @@ def from_command(
         tasks = [{"command": command, "slot": "task", "count": task_count_dict}]
         attributes = {"system": {"duration": 0}}
         return cls(resources, tasks, attributes=attributes)
+
+    @classmethod
+    def from_batch_command(
+        cls, script, args=None, num_slots=1, cores_per_slot=1, gpus_per_slot=None, num_nodes=None, broker_opts=None
+    ):
+        """Create a Jobspec describing a nested Flux instance.
+
+        Use setters to assign additional properties.
+
+        :param script: contents of the script to execute, as a string
+        :param args: arguments to pass to `script`
+        :type args: iterable of `str`
+        :param num_slots: number of resource slots to create. Slots are an abstraction,
+            and are only used (along with `cores_per_slot` and `gpus_per_slot`) to
+            determine the nested instance's allocation size and layout.
+        :param cores_per_slot: number of cores to allocate per slot
+        :param gpus_per_slot: number of GPUs to allocate per slot
+        :param num_nodes: distribute allocated resource slots across N individual nodes
+        :param broker_opts: options to pass to the new Flux broker
+        :type broker_opts: iterable of str
+        """
+
+        jobspec = cls.from_command(
+            command=["flux" *args],  # argv[0] will be replaced with the script
```

Comment:
Does this need a comma in the list construction?
```suggestion
            command=["flux", *args],  # argv[0] will be replaced with the script
```

And I am not quite following why `flux` is being prepended to the command.

Reply:
This may have been due to insufficient comments in the original `BatchCmd` implementation.

Where possible, a "batch" job constructed like this should preserve the path to the script as `argv[0]` for job provenance and also since this will be the default "job name" cached by job-info and presented in `flux-jobs`. In the case of getting a reading a script on `stdin`, `flux-mini batch` doesn't have an arg 0, so it currently just substitutes `flux` (since in a way, the batch job is really running `flux broker`).
Sorry, that should have been documented better.

The python interface here should probably allow an argv0 to be passed in, with default of `flux` or whatever.

Reply:
Thanks for clarifying, @grondo. I just assumed that `argv[0]` would be overwritten and so needed a placeholder, but I didn't realize that it was also used to determine the job name. And thanks @SteVwonder  for catching that comma bug. I'll fix both of those, add some tests, and add another function or something for getting the kind of behavior I mentioned in #3021.